src/main/java/com/lunark/lunark/moderation/controller/AccountReportController.java

```diff
@@ -35,15 +42,40 @@ public ResponseEntity<AccountReportResponseDto> getAccountReport(@PathVariable("
     public ResponseEntity<List<AccountReportResponseDto>> getAll() {
         List<AccountReport> accountReports = accountReportService.getAll();
         List<AccountReportResponseDto> reportResponseDtos = accountReports.stream()
-                .map(report -> new AccountReportResponseDto(report.getId(), report.getDate(), report.getReporter().getId(), report.getReported().getId()))
+                .map(report -> accountReportDtoMapper.toDto(report))
                 .collect(Collectors.toList());
 
         return accountReports.isEmpty() ? new ResponseEntity<>(HttpStatus.NO_CONTENT) : new ResponseEntity<>(reportResponseDtos, HttpStatus.OK );
     }
 
+    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<AccountReportResponseDto> createReport(@RequestBody AccountReportRequestDto reportRequestDto) {
```

Comment:
Does this method need a preauth?

Reply:
Yes it does. Thank you for letting me know 🙏